front/src/Api/IframeListener.ts

```diff
@@ -86,8 +86,12 @@ class IframeListener {
                 else if(payload.type === 'goToPage' && isGoToPageEvent(payload.data)) {
                     scriptUtils.goToPage(payload.data.url);
                 }
-                else if(payload.type === 'openCoWebSite' && isOpenCoWebsite(payload.data)) {
-                    scriptUtils.openCoWebsite(payload.data.url);
+                else if (payload.type === 'openCoWebSite' && isOpenCoWebsite(payload.data)) {
+                    const scriptUrl = [...this.scripts.keys()].find(key => {
+                        return this.scripts.get(key)?.contentWindow == message.source
+                    })
```

Comment:
This will work for scripts loaded with the "script" property, but will not work for scripts loaded in an iFrame (using `openWebsiteAllowApi` documented here: https://workadventu.re/map-building/scripting)

You should probably try to find the root of the iFrame in `this.iframes` if the script is not found in `this.scripts`.

Reply:
ok i also added the iframe src if the script cant be found - used the loop above